project-clone/internal/utils.go

```diff
@@ -117,3 +119,35 @@ func checkSubPathProjectState(project *dw.Project) (needClone, needRemotes bool,
 		return false, false, fmt.Errorf("could not check project state for project %s -- path %s exists and is not a directory", project.Name, clonePath)
 	}
 }
+
+func GetAdditionalCerts() (certs *x509.CertPool, warnings []string, err error) {
+	rootCAs, err := x509.SystemCertPool()
+	if err != nil {
+		return nil, warnings, fmt.Errorf("could not read system certificates pool: %w", err)
+	}
+
+	certFiles, err := os.ReadDir(publicCertsDir)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return rootCAs, warnings, nil
+		}
+		return nil, warnings, fmt.Errorf("could not read files in %s: %w", publicCertsDir, err)
+	}
+	for _, certFile := range certFiles {
+		if certFile.IsDir() {
+			continue
+		}
+		ext := filepath.Ext(certFile.Name())
+		if ext == ".crt" || ext == ".pem" {
```

Comment:
Why do we stick to specific file extensions?
All files in the directory should be certificates in a correct format.

Reply:
I did it this way for two reasons:

1. While Che only mounts certificates within the directory, other tools might not and this seemed "safer" -- I'm trying to keep the functionality as generic as I can
2. When a configmap/secret is mounted to a pod, there's other hidden files that get mounted too and I don't know what effect trying to include those might have:
    ```
    $ ls -al /public-certs
    total 0
    drwxrwsrwx. 3 root user  91 Aug  8 18:00 .
    dr-xr-xr-x. 1 root root 153 Aug  8 18:00 ..
    drwxr-sr-x. 2 root user  37 Aug  8 18:00 ..2023_08_08_18_00_15.3464882180
    lrwxrwxrwx. 1 root user  32 Aug  8 18:00 ..data -> ..2023_08_08_18_00_15.3464882180
    lrwxrwxrwx. 1 root user  30 Aug  8 18:00 kube-root-ca.crt.ca.crt -> ..data/kube-root-ca.crt.ca.crt
    ```